compiler/jsgen.nim

```diff
@@ -863,13 +863,35 @@ proc countJsParams(typ: PType): int =
 
 const
   nodeKindsNeedNoCopy = {nkCharLit..nkInt64Lit, nkStrLit..nkTripleStrLit,
-    nkFloatLit..nkFloat64Lit, nkCurly, nkPar, nkTupleConstr, nkObjConstr, nkStringToCString,
+    nkFloatLit..nkFloat64Lit, nkPar, nkStringToCString,
```

Comment:
nkCurly is used for sets and these are arrays of integers in JS, so they don't need deeop copying. I think.

Reply:
Yeah, I made it an explicit check to be a bit more future proof in case non-integer sets are ever supported, but I suppose that's unlikely to happen. So I guess you'd prefer if it to just assume that sets don't need to be copied?

Also, I guess the nkBracket case could just check if the elemType is a int/ref/ptr to avoid iterating over each expression?

Reply:
> So I guess you'd prefer if it to just assume that sets don't need to be copied?

yeah.

> Also, I guess the nkBracket case could just check if the elemType is a int/ref/ptr to avoid iterating over each expression?

I don't care either way.